### PR TITLE
Hanabi update

### PR DESCRIPTION
**Pull request by elliottower**
# Description

Splitting https://github.com/Farama-Foundation/PettingZoo/pull/933 into separate PRs for simplicity, see original for comments about changes.

Note: now that the PR has been split, this will fail until https://github.com/Farama-Foundation/PettingZoo/pull/946 is merged, as it fixes edge cases with testing that cause this to fail.

Fixes # (issue), Depends on # (pull request)

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

### Screenshots

> Please attach before and after screenshots of the change if applicable.
> To upload images to a PR -- simply drag and drop or copy paste.

# Checklist:

- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have run `pytest -v` and no errors are present.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ classic = [
     "chess==1.7.0",
     "rlcard==1.0.5",
     "pygame==2.3.0",
-    "hanabi_learning_environment==0.0.4",
+    "shimmy[openspiel]>=0.2.1"
```
Comment (jjshoots) [inline pyproject.toml:40]: The below version of shimmy uses 0.2.0.

**Review by jjshoots — CHANGES_REQUESTED** [review]
LGTM, but one small detail.

**Comment by elliottower** [comment]
Note: I changed the source distribution test to use the recommended way using setuptools: https://setuptools.pypa.io/en/latest/build_meta.html,  rather than the old way using setup.py, which they call legacy and says can produce unwanted behavior depending on systems (in my case, it seems like it doesn't work with the specific branch of shimmy, I could make a release for it and test that but I wanted to ensure the CI passes here beforehand so I'm doing this, I think it's good to update things and remove the legacy code using setup.py though so I'll probably keep this).

**Comment by elliottower** [comment]
Waiting for Shimmy release to merge this

**Pull request merged**